CMakeLists.txt

```diff
@@ -292,11 +206,11 @@ if(ENABLE_TOOLS)
     add_executable(${program} ${path})
     set_target_properties(${program} PROPERTIES FOLDER "Tools")
     create_source_groups("Source Files" ${path})
-    target_link_libraries(${program} valhalla)
-    target_include_directories(${program} PUBLIC ${VALHALLA_SOURCE_DIR}/third_party/cxxopts/include)
-    if(MSVC)
-      target_link_libraries(${program} ${CURL_LIBRARIES})
-    endif()
+    target_link_libraries(${program}
+      valhalla
+      # TODO(nils): make this a target-specific dependency? I'm sure it's not many needing this..
+      $<$<BOOL:${CURL_FOUND}>:CURL::libcurl>)
```

Comment:
looking at the sources/headers I'm quite sure libcurl is needed for some executables. I didn't really make the effort to find out which..

Reply:
it would be all of them if ENABLE_HTTP is turned on and none of them otherwise. this is because curl is used from within the valhalla library which every single one of these executables links. i would just remove the TODO